src/client/dfs/dfs.c

```diff
@@ -323,6 +340,210 @@ check_tx(daos_handle_t th, int rc)
 	return rc;
 }
 
+static int
+decode_one_hint(char *hint, uint32_t rf, daos_oclass_hints_t *obj_hint, enum daos_otype_t *type)
+{
+	char *name;
+	char *val;
+	char *save = NULL;
+
+	/** get object type (file or dir) */
+	name = strtok_r(hint, ":", &save);
+	if (name == NULL) {
+		D_ERROR("Invalid object type in hint: %s\n", hint);
+		return EINVAL;
+	}
+	if (strcasecmp(name, "dir") == 0 || strcasecmp(name, "directory") == 0) {
+		*type = DAOS_OT_MULTI_HASHED;
+		/** get hint value */
+		val = strtok_r(NULL, "", &save);
+		if (val == NULL) {
+			D_ERROR("Invalid Hint value for directory type (%s)\n", hint);
+			return EINVAL;
+		}
+		if (strcasecmp(val, "single") == 0) {
+			if (rf == 0)
+				*obj_hint = DAOS_OCH_SHD_TINY;
+			else
+				*obj_hint = DAOS_OCH_SHD_TINY | DAOS_OCH_RDD_RP;
+		} else if (strcasecmp(val, "max") == 0) {
+			if (rf == 0)
+				*obj_hint = DAOS_OCH_SHD_MAX;
+			else
+				*obj_hint = DAOS_OCH_SHD_MAX | DAOS_OCH_RDD_RP;
```

Comment:
So only replication, not EC, and we don't distinguish between RP2 or RP3? Does `daos_obj_get_oclass` do something more with this?

Reply:
we do not want to use EC for directories unless user explicitly sets that.
the redundancy level is determined later in the oclass selection according to the rf factor. these are just hints of the redundancy type.

Reply:
Ah, I mixed up file/dir. This makes sense. You did use `DAOS_OCH_RDD_EC` for files.